app/src/main/java/echoserver/SocketIo.java

```diff
@@ -0,0 +1,40 @@
+package echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class SocketIo {
+  BufferedReader in;
+  PrintWriter out;
+
+  public SocketIo() {
+    this.in = null;
```

Comment:
only need explicit `this` if you had vars with the same name. IE if you passed in `in` from the constructor then you would need it.